src/components/FormInput/FormInput.styles.ts

```diff
@@ -0,0 +1,69 @@
+import { TextField } from '@mui/material';
+import { styled } from '@mui/material/styles';
+
+import { SIZE_CONFIGS } from './FormInput.constants';
+
+import type { StyledFormInputProps } from './FormInput.types';
+
+export const StyledFormInput = styled(TextField, {
+  shouldForwardProp: (prop) => (typeof prop === 'string' ? !prop.startsWith('$') : true),
+})<StyledFormInputProps>(({ theme, $variant, $size, $error }) => {
+  const sizeConfig = $size ? SIZE_CONFIGS[$size] : SIZE_CONFIGS['medium'];
+
+  return {
+    width: '100%',
+    '& .MuiInputBase-root': {
+      height: theme.spacing(Number(sizeConfig.height.replace('rem', '')) * 2),
+      padding: theme.spacing(0),
+      '& .MuiInputBase-input': {
+        padding: theme.spacing(
+          Number(sizeConfig.padding.split(' ')[0].replace('rem', '')) * 2,
+          Number(sizeConfig.padding.split(' ')[1].replace('rem', '')) * 2
+        ),
+        fontSize: sizeConfig.fontSize,
+        '&::placeholder': {
+          color: $error ? theme.palette.error.main : theme.palette.divider,
+          opacity: 1,
+        },
+      },
+    },
+    '& .MuiOutlinedInput-root': {
+      '& fieldset': {
+        borderColor: $error ? theme.palette.error.main : theme.palette.divider,
+        borderRadius: theme.shape.borderRadius,
+        borderWidth: '1px',
+      },
+      '&:hover fieldset': {
+        borderColor: $error ? theme.palette.error.main : theme.palette.custom?.focus,
+        borderWidth: '1px',
+      },
+      '&.Mui-focused fieldset': {
+        borderColor: theme.palette.custom?.focus,
+        borderWidth: '1px',
+      },
+      ...($variant === 'filled' && {
```

Comment:
Using a computed spread with ...($variant === 'filled' && { ... }) will spread 'false' when $variant !== 'filled' which actually results in no-op, but the pattern is harder to read and may confuse. Additionally, inside you remove the fieldset border (border: 'none') while leaving other selectors intact. Verify that for the 'filled' variant the MUI classnames used here are correct and consistent.

```typescript
Prefer a clear conditional object merge or ternary to return an object. Example:

...($variant === 'filled' ? {
  '& fieldset': { border: 'none', backgroundColor: theme.palette.action.selected }
} : {}),

Also confirm the correct MUI classes for 'filled' variant.
```